dp.md

```diff
@@ -0,0 +1,19 @@
+Dynamic Programming
+===================
+Dynamic Programming (DP) is to plan the best solution dynamically by using the
+result of solved subproblems. When we are solving DP problems, we need to think
+about the recurrence relation: how is knowing the results of solved subproblems
+going to help.
+
+
+DP problems can be solved recursively or iteratively. The iterative solutoins
+usually have a performance advantage, however, the overhead of the recursive
```

Comment:
```suggestion
usually has a performance advantage, however, the overhead of the recursive
```

Reply:
Hmm, maybe it should be `The iterative solutoins have performance advantages`?

Reply:
I figured it would be `The iterative solution has performance advantages`, but now that I reread it, it sounds good both ways.